webrtc-c/webrtc-and-directstream/kvsWebRTCAndDirectStream.c

```diff
@@ -0,0 +1,409 @@
+                    printf("writeFrame failed with 0x%08x\n", status);
+                } else if (status == STATUS_SUCCESS && pSampleStreamingSession->firstFrame) {
+                    pSampleStreamingSession->firstFrame = FALSE;
+                    printf("First frame sent successfully\n");
+                }
+            }
+            MUTEX_UNLOCK(pSampleConfiguration->streamingSessionListReadLock);
+
+            gst_buffer_unmap(buffer, &map);
+        }
+    }
+
+    gst_sample_unref(sample);
+    return GST_FLOW_OK;
+}
+
+int main(int argc, char *argv[]) {
+    STATUS retStatus = STATUS_SUCCESS;
+    PSampleConfiguration pSampleConfiguration = NULL;
+    SignalingClientMetrics signalingClientMetrics;
+    PCHAR pChannelName;
+    PCHAR pStreamName;
+    GstElement *source, *capsfilter, *convert, *tee;
+    
+    // WebRTC branch elements
+    GstElement *webrtc_queue, *webrtc_convert, *webrtc_encoder, *webrtc_caps, *webrtc_appsink;
+    
+    // KVS branch elements
+    GstElement *kvs_queue, *kvs_convert, *kvs_encoder, *kvs_parse, *kvs_caps, *kvs_sink;
+    
+    GstBus *bus;
+    GstPad *tee_webrtc_pad = NULL, *tee_kvs_pad = NULL;
+    GstPad *webrtc_queue_pad, *kvs_queue_pad;
+    GstCaps *caps, *webrtc_h264_caps, *kvs_h264_caps;
+
+    // Initialize GStreamer first
+    gst_init(&argc, &argv);
+
+    // Check arguments
+    if (argc < 3) {
+        g_printerr("Usage: %s <signaling-channel> <stream-name>\n", argv[0]);
+        return 1;
+    }
+
+    pChannelName = argv[1];
+    pStreamName = argv[2];
+
+    // Set up signal handlers
+    signal(SIGINT, signal_handler);
+    signal(SIGTERM, signal_handler);
+
+    // Create the main loop
+    main_loop = g_main_loop_new(NULL, FALSE);
+
+    // Create WebRTC sample configuration - video only, no audio
+    retStatus = createSampleConfiguration(pChannelName, SIGNALING_CHANNEL_ROLE_TYPE_MASTER, TRUE, FALSE, TRUE, &pSampleConfiguration);
+    if (retStatus != STATUS_SUCCESS) {
+        printf("createSampleConfiguration(): operation returned status code: 0x%08x\n", retStatus);
+        goto CleanUp;
+    }
+
+    // Enable media storage
+    pSampleConfiguration->channelInfo.useMediaStorage = TRUE;
+    
+    // Configure for better connectivity
+    pSampleConfiguration->trickleIce = TRUE;
+    pSampleConfiguration->useTurn = TRUE;
+    
+    // Set media type to video only
+    pSampleConfiguration->mediaType = SAMPLE_STREAMING_VIDEO_ONLY;
+    
+    // Set custom data for callbacks
+    pSampleConfiguration->customData = (UINT64) pSampleConfiguration;
+    
+    printf("Created signaling channel %s\n", pChannelName);
+    printf("ICE configuration: trickleIce=%s, useTurn=%s\n", 
+           pSampleConfiguration->trickleIce ? "enabled" : "disabled",
+           pSampleConfiguration->useTurn ? "enabled" : "disabled");
+
+    // Set KVS log configuration path
+    setenv("KVS_LOG_CONFIG", "./kvs_log_configuration", 1);
```

Comment:
Is the path correct if we're running from the `amazon-kinesis-video-streams-webrtc-sdk-c/build/` location mentioned in the README?